gradient_descent_2D.ipynb

```diff
@@ -296,71 +296,89 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 8,
+   "execution_count": 52,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "134: R_E= 1.3673e-06, ratio= 7.2449e+00, output= 1070640.217\n",
-      "578: R_E= 3.0204e-06, ratio= 6.1429e+00, output= 531759.877\n",
+      "134: R_E= 1.3673e-06, ratio= 7.2449e+00, output= 6.018\n",
+      "578: R_E= 3.0204e-06, ratio= 6.1429e+00, output= 5.705\n",
       "67 missing values fixed\n"
      ]
     }
    ],
    "source": [
-    "# Interpolate missing values:\n",
-    "\n",
-    "vmin = min(output_list)\n",
-    "vmax = 100.0*vmin\n",
-    "combined = [(x, y) for x, y in zip(list1, list2)]\n",
+    "# Interpolate missing values\n",
```

Comment:
2D contour plots in the example have a feature to interpolate missing values and remove outliers. 
The code is better commented, and works better. (Actually, I believe, that the previous version only worked with my example) 
Non-logaritmic scale is supported. (use_log10: bool)